core/libp2p/transport/asio/asio_app.hpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_ASIO_APP_HPP
+#define KAGOME_ASIO_APP_HPP
+
+#define BOOST_ASIO_NO_DEPRECATED
+
+#include <boost/asio.hpp>
+#include <boost/thread.hpp>
+
+namespace libp2p::transport::asio {
+
+  /**
+   * @brief Class, which owns boost::asio::io_context, thread pool. Should be 1
+   * per application.
+   */
+  class AsioApp : private boost::asio::noncopyable {
```

Comment:
Why boost::asio::non copyable ?
AFAIK, boost::noncopyable fits as well.
[boost::noncopyable](https://www.boost.org/doc/libs/1_65_0/libs/core/doc/html/core/noncopyable.html#core.noncopyable.header_boost_core_noncopyable_hp)
What is the purpose?

Reply:
IDE showed me boost asio noncopyable :)